### PR TITLE
Do not error when there is no response content but accept header is set 

### PR DESCRIPTION
**Pull request by ilanashapiro**
Addresses #2266

**Summary**

We want to mock test a customer's method called with accept: application/json even though the method doesn't respond with a body. Currently, Prism complains that it doesn't know what response body to produce and throws an error. Now, if the spec defines a response body, but does NOT have a media type defined for the one requested with the accept header, Prism still throws an error. However, if the spec does NOT define a response body for the endpoint, we essentially ignore the accept header, and do not throw an error even if the media type doesn't match. 

**Checklist**

- The basics
  - [X] I tested these changes manually in my local or dev environment
- Tests
  - [X] Added or updated
  - [ ] N/A
- Event Tracking
  - [ ] I added event tracking and followed the event tracking guidelines
  - [X] N/A
- Error Reporting
  - [ ] I reported errors and followed the error reporting guidelines
  - [X] N/A

**Screenshots**

Example OpenAPI spec used for end-end testing with content (file is "example2.yaml"):
image

Request with no accept headers:
image

image

Request with accept header matching content media type:
image

image

Request with accept header that does NOT match content media type:
image

image

Example OpenAPI spec used for end-end testing with NO content (file is still "example2.yaml"):
image

Request with no accept headers:
image

image

Request with accept header:
image

image

### CODE REVIEW

**Comment by ilanashapiro** [comment]
@rattrayalex

**Comment by chohmann** [comment]
@ilanashapiro thank you for the PR! As mentioned in the [corresponding issue](https://github.com/stoplightio/prism/issues/2266#issuecomment-1518064570), we think this should just be how Prism behaves all the time, rather than have it be configurable. Do you want to take a crack at updating this PR to accomplish this?

**Comment by ilanashapiro** [comment]
@chohmann @brendarearden

File: packages/http/src/mocker/negotiator/NegotiatorHelpers.ts [hunk 1]
```diff
@@ -50,7 +50,7 @@ const helpers = {
     httpContent: IMediaTypeContent
   ): E.Either<Error, BodyNegotiationResult> {
     const { mediaType } = httpContent;
-
+
```
Comment (chohmann) [inline packages/http/src/mocker/negotiator/NegotiatorHelpers.ts:53]:
```suggestion

```

**Review by chohmann — APPROVED** [review]

**Comment by chohmann** [comment]
@ilanashapiro thank you for your contribution!

**Comment by chohmann** [comment]
@ilanashapiro this was released in version 4.13.0 of Prism. Thanks again for your contribution!

**Pull request merged**